### PR TITLE
Refactor Makefile to reduce copy-pasted rules

### PR DESCRIPTION
**Pull request by nullpo-head**
**What this PR does / why we need it**:

This PR refactors `Makefile.core.mk` to reduce the duplicated copy-and-pasted rules. 
Currently, the target variations of `build*` and `test*` are written by copying and pasting the original targets, which makes them less maintainable. You had to modify all targets without mistake when you changed the rules.

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -119,33 +123,29 @@ gen:
...
+test_asan: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_ASAN)
+test_asan: test
 
-test_tsan: V8_BUILD_CONFIG=$(BAZEL_CONFIG_TSAN)
-test_tsan: prerun_v8_build
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_TSAN) //src/envoy:envoy
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_TSAN) -- $(BAZEL_TEST_TARGETS)
-	env ENVOY_PATH=$(BAZEL_ENVOY_PATH) TSAN=true GO111MODULE=on go test -timeout 30m ./...
+test_tsan: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_ASAN)
```
Comment (bianpengyuan) [inline Makefile.core.mk:139]: BAZEL_CONFIG_TSAN?

File: Makefile.core.mk [hunk 2]
```diff
@@ -119,33 +123,29 @@ gen:
...
+test_tsan: test
+
+test_centos: BAZEL_BUILD_ARGS := $(CENTOS_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT)
```
Comment (bianpengyuan) [inline Makefile.core.mk:142]: Should concat with `BAZEL_BUILD_ARGS` instead of `BAZEL_CONFIG_CURRENT `?

File: Makefile.core.mk [hunk 3]
```diff
@@ -119,33 +123,29 @@ gen:
...
+test_centos: E2E_TEST_TARGETS =
+test_centos: BAZEL_TEST_TARGETS = $(CENTOS_BAZEL_TEST_TARGETS)
+# TODO: re-enable IPv6 tests
+test_centos: BAZEL_TEST_ARGS = --test_filter="-*IPv6*"
+test_centos: test
```
Comment (bianpengyuan) [inline Makefile.core.mk:147]: I looked at the presubmit job and seems like start up args are not applied to test_centos: https://prow.istio.io/log?container=test&id=1473748096448991232&job=release-centos-test_proxy Could you please check on that?

File: Makefile.core.mk (hunk 1 above)
Comment (nullpo-head) [reply Makefile.core.mk]: Fixed. Thanks for the catch!

File: Makefile.core.mk [hunk 4]
```diff
@@ -73,28 +76,29 @@ CENTOS_BAZEL_TEST_TARGETS ?= ${BAZEL_TARGETS} -tools/deb/... -tools/docker/... -
...
 
-build_envoy_asan: V8_BUILD_CONFIG=$(BAZEL_CONFIG_ASAN)
-build_envoy_asan: prerun_v8_build
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_ASAN) //src/envoy:envoy
+build_envoy_asan: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_ASAN)
+build_envoy_asan: BAZEL_TARGETS = //src/envoy:envoy
+build_envoy_asan: build
+
+build_envoy_centos: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_REL)
```
[the remark this answers is not in the document]
Comment (nullpo-head) [reply Makefile.core.mk]: OK, I've removed it. I just added the target to build centos for consistency. There are targets to test the debug build, asan, tsan, and centos, but there were only targets to build the debug, asan and tsan, and there was no centos.

File: Makefile.core.mk (hunk 2 above)
Comment (nullpo-head) [reply Makefile.core.mk]: Thanks for the catch. I've fixed it.

File: Makefile.core.mk (hunk 3 above)
Comment (nullpo-head) [reply Makefile.core.mk:142]: It looks the cause was the test_centos's rule. It should be fixed now.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**